test/sys/test_socket.rs

```diff
@@ -220,12 +223,84 @@ mod recvfrom {
             SockFlag::empty(),
             None,
         ).expect("send socket failed");
-        let from = sendrecv(rsock, ssock, move |s, m, flags| {
+        let from = sendrecv(rsock, ssock, None, move |s, m, flags| {
             sendto(s, m, &sock_addr, flags)
         });
         // UDP sockets should set the from address
         assert_eq!(AddressFamily::Inet, from.unwrap().family());
     }
+
+    #[cfg(target_os = "linux")]
+    mod udp_offload {
+        use super::*;
+        use nix::sys::uio::IoVec;
+        use nix::sys::socket::sockopt::{UdpGroSegment, UdpGsoSegment};
+
+        #[test]
+        pub fn gso() {
+            require_kernel_version!(udp_offload::gso, ">= 4.18");
+
+            let std_sa = SocketAddr::from_str("127.0.0.1:6791").unwrap();
+            let inet_addr = InetAddr::from_std(&std_sa);
+            let sock_addr = SockAddr::new_inet(inet_addr);
+            let rsock = socket(AddressFamily::Inet,
+                               SockType::Datagram,
+                               SockFlag::empty(),
+                               None
+            ).unwrap();
+
+            setsockopt(rsock, UdpGsoSegment, &2).expect("setsockopt UDP_SEGMENT failed");
```

Comment:
This `2` is a magic number.  Please assign it to a named variable, and reference that variable here and on line 262.